manifests/config.pp

```diff
@@ -22,18 +25,23 @@
         # The standard packages install alternatives, custom packages do not
         # For the stanard packages java::params needs these added.
         if $java::use_java_package_name != $java::default_package_name {
+          $command_redhat = ['alternatives', '--install', '/usr/bin/java', 'java', $java::use_java_alternative_path, '20000']
+          $unless_redhat = ["alternatives --display java | grep -q ${java::use_java_alternative_path}"]
```

Comment:
This relies on a shell, so perhaps explicitly keep it a string to make that clear?
```suggestion
          $unless_redhat = "alternatives --display java | grep -q ${java::use_java_alternative_path}"
```

Then you should escape the value.